src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -1277,81 +1252,24 @@ private MethodSymbol GetFixedPatternMethodOpt(BoundExpression initializer, Diagn
-                    initializer.Syntax,
-                    initializer,
-                    methodName,
-                    rightArity: 0,
-                    typeArgumentsSyntax: default,
-                    typeArguments: default,
-                    invoked: true,
-                    indexed: false,
-                    bindingDiagnostics);
-
-                if (boundAccess.Kind != BoundKind.MethodGroup)
-                {
-                    // the thing is not even a method
-                    return null;
-                }
-
-                var analyzedArguments = AnalyzedArguments.GetInstance();
-                BoundExpression patternMethodCall = BindMethodGroupInvocation(
-                    initializer.Syntax, 
-                    initializer.Syntax, 
-                    methodName, 
-                    (BoundMethodGroup)boundAccess, 
-                    analyzedArguments, 
-                    bindingDiagnostics, 
-                    queryClause: null, 
-                    allowUnexpandedForm: false);
-
-                analyzedArguments.Free();
-
-                if (patternMethodCall.Kind != BoundKind.Call)
-                {
-                    // did not find anything callable
-                    return null;
-                }
-
-                var call = (BoundCall)patternMethodCall;
-                if (call.ResultKind == LookupResultKind.Empty)
-                {
-                    // did not find any methods that even remotely fit
-                    return null;
-                }
-
-                // we have succeeded or almost succeded to bind the method
-                // report additional binding diagnostics that we have seen so far
-                additionalDiagnostics.AddRange(bindingDiagnostics);
+            var result = FindPatternMethodRelaxed(initializer, methodName, initializer.Syntax, additionalDiagnostics, out var patterMethodSymbol);
 
-                var patterMethodSymbol = call.Method;
-                if (patterMethodSymbol is ErrorMethodSymbol ||
-                    patternMethodCall.HasAnyErrors)
-                {
-                    // bound to something uncallable
-                    return null;
-                }
-
-                if (HasOptionalOrVariableParameters(patterMethodSymbol) ||
-                    patterMethodSymbol.ReturnsVoid ||
-                    !patterMethodSymbol.RefKind.IsManagedReference() ||
-                    !(patterMethodSymbol.ParameterCount == 0 || patterMethodSymbol.IsStatic && patterMethodSymbol.ParameterCount == 1))
-                {
-                    // the method does not fit the pattern
-                    additionalDiagnostics.Add(ErrorCode.WRN_PatternBadSignature, initializer.Syntax.Location, initializer.Type, "fixed", patterMethodSymbol);
-                    return null;
-                }
-
-                return patterMethodSymbol;
+            if (result != PatternLookupResult.Success)
```

Comment:
Consider replacing this with `(object)patternMethodSymbol != null` in the `if` below. That way it's clear that `patternMethodSymbol` is not null when it is dereferenced.